### PR TITLE
table designer improvements

### PR DESCRIPTION
**Pull request by alanrenmsft**
1. vBump sts to bring in the new table designer changes
2. update publish changes operation to return updated table info (e.g. isNewTable will become false after saving)
3. post processing of publish changes operation (e.g. update designer table label to reflect the new name)
3. support confirmation dialog when removing items from designer grid
4. fix the name of a few properties.

### CODE REVIEW

**Review by abist — APPROVED** [review]

File: src/sql/azdata.proposed.d.ts [hunk 1]
```diff
@@ -1238,7 +1238,7 @@ declare module 'azdata' {
 			indexColumnSpecificationTableOptions?: TableDesignerBuiltInTableViewOptions;
 		}
 
-		export interface TableDesignerBuiltInTableViewOptions {
+		export interface TableDesignerBuiltInTableViewOptions extends DesignerTablePropertiesBase {
```
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:1241]: Any reason for having most of the classes be `TableDesigner...` but the new one be `DesignerTable...`?

File: src/sql/azdata.proposed.d.ts [hunk 2]
```diff
@@ -1317,39 +1309,45 @@ declare module 'azdata' {
...
+			/**
+			 * Whether user can remove rows from the table. The default value is true.
+			 */
+			canRemoveRows?: boolean;
+			/**
+			 * Whether to show confirmation when user removes a row.
+			 */
+			showRemoveRowConfirmation?: boolean;
+			/**
+			 * The confirmation message to be displayed when user removes a row.
```
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:1326]: Indicate default for these other optional props

File: src/sql/workbench/api/common/sqlExtHostTypes.ts [hunk 3]
```diff
@@ -969,15 +969,15 @@ export namespace designers {
...
+		ForeignTable = 'foreignTable',
 		OnDeleteAction = 'onDeleteAction',
 		OnUpdateAction = 'onUpdateAction',
 		Columns = 'columns'
 	}
 
 	export enum ForeignKeyColumnMappingProperty {
-		PrimaryKeyColumn = 'primaryKeyColumn',
-		ForeignKeyColumn = 'foreignKeyColumn'
+		column = 'column',
```
Comment (Charles-Gagnon) [inline src/sql/workbench/api/common/sqlExtHostTypes.ts:979]: These enum names should be capitalized

File: src/sql/workbench/browser/designer/designer.ts [hunk 4]
```diff
@@ -303,11 +305,22 @@ export class Designer extends Disposable implements IThemable {
...
+							// Ignore the slick grid error when setting active cell.
+						}
+					} else {
+						this.updatePropertiesPane(this._propertiesPane.objectPath);
 					}
 				} else if (edit.type === DesignerEditType.Update) {
 					// for edit, update the properties pane with new values of current object.
 					this.updatePropertiesPane(this._propertiesPane.objectPath);
+				} else {
+					if (edit.path.length === 4) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/designer/designer.ts:321]: Comment explaining what's special about 4

File: src/sql/workbench/browser/designer/designer.ts [hunk 5]
```diff
@@ -693,7 +706,16 @@ export class Designer extends Disposable implements IThemable {
 						resizable: false,
 						isFontIcon: true
 					});
-					deleteRowColumn.onClick((e) => {
+					deleteRowColumn.onClick(async (e) => {
+						if (tableProperties.showRemoveRowConfirmation && tableProperties.removeRowConfirmationMessage) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/designer/designer.ts:710]: Consider having some error checking somewhere if the user has `showRemoveRowConfirmation` set but not `removeRowConfirmationMessage`

Alternatively you could make these two properties part of a single object `rowConfirmationOptions` that's optional itself, but then these properties are both required inside of it if it's included.

File: src/sql/workbench/browser/designer/interfaces.ts [hunk 6]
```diff
@@ -179,6 +175,14 @@ export interface DesignerTableProperties extends ComponentProperties {
 	 * Whether user can remove rows from the table. The default value is true.
 	 */
 	canRemoveRows?: boolean;
+	/**
+	 * Whether to show confirmation when user removes a row.
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/designer/interfaces.ts:179]: Default values in js doc

**Review by Charles-Gagnon — APPROVED** [review]

File: src/sql/azdata.proposed.d.ts (hunk 1 above)
Comment (alanrenmsft) [reply src/sql/azdata.proposed.d.ts:1241]: :) the newly added interface is table component properties base interface for designer.

File: src/sql/workbench/browser/designer/designer.ts (hunk 5 above)
Comment (alanrenmsft) [reply src/sql/workbench/browser/designer/designer.ts:710]: I'll have a default remove row confirmation message.

**Review by caohai — APPROVED** [review]

**Pull request merged**